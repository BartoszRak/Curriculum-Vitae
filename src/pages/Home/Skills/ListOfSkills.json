[
  {
    "name": "React",
    "advancement": 0
  }, {
    "name": "Testing Library",
    "advancement": 0
  }, {
    "name": "Javascript",
    "advancement": 0
  }, {
    "name": "Material UI",
    "advancement": 0
  }, {
    "name": "Express",
    "advancement": 0
  }, {
    "name": "Nestjs",
    "advancement": 0
  }, {
    "name": "Scss/Sass",
    "advancement": 0
  }, {
    "name": "Axios",
    "advancement": 0
  }, {
    "name": "Firebase",
    "advancement": 0
  }, {
    "name": "TypeScript",
    "advancement": 0
  }, {
    "name": "Rematch",
    "advancement": 0
  }, {
    "name": "Redux",
    "advancement": 0
  }, {
    "name": ".Jss",
    "advancement": 0
  }, {
    "name": "Immutable",
    "advancement": 0
  }, {
    "name": "Git",
    "advancement": 0
  }, {
    "name": "Formik",
    "advancement": 0
  }, {
    "name": "HTML",
    "advancement": 0
  }, {
    "name": "CSS",
    "advancement": 0
  }, {
    "name": "Rest API",
    "advancement": 0
  }, {
    "name": "Travis CI",
    "advancement": 1
  }, {
    "name": "Keycloak",
    "advancement": 1
  }, {
    "name": "Webpack",
    "advancement": 1
  }, {
    "name": "Sinon",
    "advancement": 1
  }, {
    "name": "MongoDB",
    "advancement": 1
  }, {
    "name": "Next.js",
    "advancement": 1
  }, {
    "name": "Typescript REST",
    "advancement": 1
  }, {
    "name": "GraphQL",
    "advancement": 1
  }, {
    "name": "jQuery",
    "advancement": 1
  }, {
    "name": "C++",
    "advancement": 2
  }, {
    "name": "SFML v2.0",
    "advancement": 2
  }, {
    "name": "Boost",
    "advancement": 2
  }, {
    "name": "Golang",
    "advancement": 2
  }, {
    "name": "Python",
    "advancement": 2
  }, {
    "name": "Gulp",
    "advancement": 2
  }, {
    "name": "Styled Components",
    "advancement": 2
  }, {
    "name": "Php",
    "advancement": 2
  }, {
    "name": "MySQL",
    "advancement": 2
  }, {
    "name": "Animejs",
    "advancement": 2
  }
]