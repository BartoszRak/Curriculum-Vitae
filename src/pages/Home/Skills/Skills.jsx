import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core'

import style from './Skills.style'

export function Skills({ classes }) {
  return (
    <div className={classes.root} id="skills">
      skills
    </div>
  )
}

Skills.propTypes = {
  classes: PropTypes.object,
}

export default withStyles(style)(Skills)