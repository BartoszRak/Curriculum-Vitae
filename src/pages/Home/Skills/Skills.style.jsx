export default ({ breakpoints, spacing, palette }) => ({
  emoji: {
    fontSize: '2rem',
    [breakpoints.down('md')]: {
      fontSize: '1.5rem',
    },
  },
  root: {
    width: '100%',
  },
  section: {},
  skills: {
    marginTop: 20,
  },
  subtitle: {
    borderLeft: `4px solid ${palette.secondary.main}`,
    marginTop: 50,
    paddingLeft: spacing.unit * 2,
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
  },
  title: {
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
    [breakpoints.down('md')]: {
      fontSize: '2rem',
    },
  },
})
