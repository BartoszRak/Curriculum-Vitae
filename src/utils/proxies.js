/* eslint-disable */
export const createPropertyProxy = (obj = {}) => new Proxy(obj, {
  get: (target, name) => target[name] || `${name}Mock`,
})

export const PropertyProxy = createPropertyProxy()
