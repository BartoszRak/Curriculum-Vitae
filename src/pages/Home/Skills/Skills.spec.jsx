import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { Skills } from './Skills'
import { PropertyProxy } from '~utils/proxies'

jest.unmock('@material-ui/core')

jest.mock('./SkillsList', () => 'SkillsListMock')

let props

describe('component: Skills', () => {
  afterEach(cleanup)

  beforeEach(() => {
    props = {
      classes: PropertyProxy,
    }
  })
  describe('rendering', () => {
    test('render without crash', () => {
      const wrapper = shallow(<Skills {...props} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot', () => {
      const wrapper = shallow(<Skills {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
