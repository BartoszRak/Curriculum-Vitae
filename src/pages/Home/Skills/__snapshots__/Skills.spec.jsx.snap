// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Skills should render as expected 1`] = `
<div
  id="skills"
>
  skills
</div>
`;
